stage #2.5/module #4 REST Basics - Spring Web/#6.REST API Tools for Testing, Design, and Development/swagger_for_designing_and_documenting_rest_api.md

```diff
@@ -0,0 +1,201 @@
+
+For each of the REST endpoint and its associated operations, You can provide ApiOperation and their various responses with ApiResponses annotations.
+
+```Java
+@RestController
+@RequestMapping("/api/v1/news")
+@Api(produces = "application/json", value = "Operations for creating, updating, retrieving and deleting news in the application")
+public class NewsController {
+
+    @Autowired
+    private NewsRepository newsRepository;
+
+    @PostMapping
+    @ApiOperation(value = "Create a piece of news", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully created a piece of news"),
+            @ApiResponse(code = 401, message = "You are not authorized to view the resource"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "The resource you were trying to reach is not found"),
+            @ApiResponse(code = 500, message = "Application failed to process the request")
+    }
+    )
+    public ResponseEntity<News> createNews(@RequestBody News news){
+       
+            News savedNews = newsRepository.save(news);
+            return new ResponseEntity<News>(savedNews, HttpStatus.OK);
+       
+    }
+
+    @GetMapping
+    @ApiOperation(value = "View all news", response = Iterable.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved all news"),
+            @ApiResponse(code = 401, message = "You are not authorized to view the resource"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "The resource you were trying to reach is not found"),
+            @ApiResponse(code = 500, message = "Application failed to process the request")
+    }
+    )
+    public Iterable<Donor> getNews(){
+        return newsRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Retrieve specific news with the supplied id", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved the news with the supplied id"),
+            @ApiResponse(code = 401, message = "You are not authorized to view the resource"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "The resource you were trying to reach is not found"),
+            @ApiResponse(code = 500, message = "Application failed to process the request")
+    }
+    )
+    public ResponseEntity<News> getNews(@PathVariable("id") Long id){
+        Optional news = newsRepository.findById(id);
```

Comment:
Raw 'Optional' usage is not recommended, `if / get` may be replaced with less verbose `Optional#orElseGet` method